rust/src/lib/nm/nm_dbus/connection/ip.rs

```diff
@@ -167,6 +168,9 @@ impl TryFrom<DbusDictionary> for NmSettingIp {
             setting.dns = _from_map!(v, "dns", parse_nm_dns)?;
         }
 
+        if v.contains_key("forwarding") {
+            setting.forwarding = _from_map!(v, "forwarding", i32::try_from)?;
```

Comment:
The `_from_map!` already has stuff like `contains_key()`, please move this `_from_map!()` to creation of `NmSettingIp` instead of modify after creation.

Reply:
Thanks, I changed the code as you suggested.